src/renderers/shared/fiber/ReactFiberStateQueue.js

```diff
@@ -37,16 +46,17 @@ exports.addToQueue = function(queue : StateQueue, partialState : mixed): StateQu
   return queue;
 };
 
-exports.mergeStateQueue = function(prevState : any, props : any, queue : ?StateQueue) : any {
-  if (!queue) {
+exports.mergeStateQueue = function(queue : ?StateQueue, prevState : any, props : any) : any {
+  let node : ?StateQueueNode = queue;
+  if (!node) {
     return prevState;
   }
   let state = Object.assign({}, prevState);
   do {
-    const partialState = typeof queue.partialState === 'function' ?
-      queue.partialState(state, props) :
-      queue.partialState;
+    const partialState = typeof node.partialState === 'function' ?
+      node.partialState(state, props) :
```

Comment:
This will pass the `node` to user code, leaking this info. Pick off the function before invoking it so that `this` will be undefined (or global depending on strict mode).

```js
const stateUpdater = node.partialState;
const partialState = stateUpdater(state, props);
```